app/assets/javascripts/school_administrator_dashboard/discipline_dashboard/SchoolDisciplineDashboard.js

```diff
@@ -46,7 +46,9 @@ class SchoolDisciplineDashboard extends React.Component {
   studentDisciplineIncidentCounts(incidentCategory) {
     let studentDisciplineIncidentCounts = {};
     const selectedChartData = this.getChartData(this.state.selectedChart);
-    const incidents = incidentCategory ? selectedChartData.disciplineIncidents[incidentCategory] : this.props.schoolDisciplineEvents;
+    const incidents = incidentCategory ?
+                      selectedChartData.disciplineIncidents[incidentCategory] :
+                      this.filterIncidentDates(this.props.schoolDisciplineEvents);
```

Comment:
@edavidsonsawyer This seems like it could work, but then it also seems like it doesn't fix the root problem, which is keeping the data in the table in sync with the data in the charts.

I'm wondering if moving the filtering into the `render` method, and then have it pass the filtered results down to the subsequent `renderStudentDisciplineTable` and `renderDisciplineChart` methods, if this would remove this class of bugs altogether.

So to sketch that:
```
render() {
  const students = this.filterStudentsBasedOnSelections(selectedChart, selectedCategory);
  const incidents = this.filterDisciplineIncidentsBasedOnSelections(selectedChart, selectedCategory);
  return (
     ...
     {this.renderStudentsTable(students, incidents)}
     {this.renderCharts(students, incidents)}
     ...
  );
```

I'm not sure if this is worthwhile, so I think that's your call to make, and since this fixes the immediate bug let's 🚢 it now :)

Reply:
@edavidsonsawyer In testing this, this PR has a bug so I reverted shipping it in this revert PR: https://github.com/studentinsights/studentinsights/pull/1816.

To reproduce, select a category (eg, "Time"), click a time of day, then change the time filter.  This is what I see locally:

<img width="902" alt="screen shot 2018-06-18 at 12 39 01 pm" src="https://user-images.githubusercontent.com/1056957/41549792-f0873f74-72f4-11e8-9df4-d0516d2c633c.png">

So we can't ship this PR as-is.  I'd recommend a refactoring similar to what I suggested [here](https://github.com/studentinsights/studentinsights/pull/1815#discussion_r196145564) as a path forward, since this class of bug seems to be making it tricky to make changes, and changing that factoring might help.  If you want to chat about adding another test or two to help catch these kinds of things, I'm happy to chat about that too.